net/host.go

```diff
@@ -220,3 +220,24 @@ func readStreamMessage(stream libp2pnetwork.Stream, maxMessageSize uint32) (comm
 
 	return message.DecodeMessage(msgBytes)
 }
+
+// nextStreamMessage returns a channel that will receive the next message from the stream.
+// if there is an error reading from the stream, the channel will be closed, thus
+// the received value will be nil.
+func nextStreamMessage(stream libp2pnetwork.Stream, maxMessageSize uint32) <-chan common.Message {
+	ch := make(chan common.Message)
+	go func() {
+		for {
+			msg, err := readStreamMessage(stream, maxMessageSize)
+			if err != nil {
+				log.Warnf("failed to read stream message: %s", err)
```

Comment:
There is at least one place this method is being called, where it is not an error for `readStreamMessage` to return `io.EOF`.  Our underlying `p2pnet.ReadStreamMessage` method already differentiates between `io.EOF` (stream closed between messages) and `io.ErrUnexpectedEOF` (truncated message).

I didn't research how to fix the issue. Maybe we don't write a warning here if the error is `io.EOF`, and we ensure that any channel reader logs a good message when the lack of response is an error?

Reply:
I updated this to not log anything and just return if the errors is `io.EOF`. another option would be to have this function return a channel for any errors that happen, which the caller can read from. what do  you think?

Reply:
I think the way you did it is good. Only the channel message reader has the context to know if the absence of a next message is an error or not. The reader's job is probably simpler without the 2nd channel since, from a protocol standpoint, they don't really care what the specific error was for their decision logic.


knows if the absence of a next message is an error or not.